File: game-server/src/main/java/org/dm/event/Event.java
package org.dm.event;

import com.dm.content.gambling.GambleStage;
import com.dm.game.world.entity.mob.player.Player;

/**
 * @author Jire
 */
public interface Event {

    /**
     * Determines if the player is in a state where they can handle this event.
     * * @param player The player to check.
     * @return true if the event can be handled, false otherwise.
     */
    default boolean canHandle(Player player) {
        return !player.isDead()
                && player.getGambling().getStage() != GambleStage.PLACING_BET
                && player.getGambling().getStage() != GambleStage.IN_PROGRESS;
    }

    /**
     * Handles the logic for this event.
     * * @param player The player involved in the event.
     */
    default void handle(Player player) {
        // Default implementation does nothing
    }

}

File: game-server/src/main/kotlin/org/jire/tarnishps/event/Event.kt
package org.jire.tarnishps.event

import com.dm.content.gambling.GambleStage
import com.dm.game.world.entity.mob.player.Player

/**
 * @author Jire
 */
interface Event {

    fun canHandle(player: Player) =
        !player.isDead
                && player.gambling.stage != GambleStage.PLACING_BET
                && player.gambling.stage != GambleStage.IN_PROGRESS

    fun handle(player: Player) {}

}